File: test/fixtures/simple.go
package main

import "fmt"

func main() {
	result := processInput("hello")
	fmt.Println(result)
}

func processInput(input string) string {
	validated := validate(input)
	return transform(validated)
}

func validate(input string) string {
	if len(input) == 0 {
		panic("empty input")
	}
	return input
}

func transform(input string) string {
	return input + "_processed"
}

File: test/fixtures/simple.rs
fn main() {
    let data = read_input();
    let result = process(data);
    output(result);
}

fn read_input() -> String {
    String::from("hello")
}

fn process(data: String) -> String {
    let validated = validate(&data);
    transform(validated)
}

fn validate(data: &str) -> &str {
    if data.is_empty() {
        panic!("empty input");
    }
    data
}

fn transform(data: &str) -> String {
    data.to_uppercase()
}

fn output(result: String) {
    println!("{}", result);
}
